### PR TITLE
OpCode::EditorOpen: return 1 if editor opened successfully

### PR DESCRIPTION
**Pull request by crsaracco**
`dispatch()` should return 1 if the editor was opened successfully. Some DAWs ignore this return value, but others don't.

One example of a DAW that doesn't is Carla ([see this chunk of code](https://github.com/falkTX/Carla/blob/51f20736b4e0b5d6416f2b7d4051f155f5858fbf/source/backend/plugin/CarlaPluginVST2.cpp#L513)).

Since we return 0 here by default, Carla assumes that the editor failed to open.

I think the more correct way to do this would be to actually check that `editor.open()` was successful in some way, but that would require an API change.

### CODE REVIEW

**Comment by askeksa** [comment]
I would suggest that we change the editor API so `editor.open()` returns a `bool` indicating success. This seems to be a method where it is relevant to report failure (even though some DAWs ignore it).

**Comment by crsaracco** [comment]
Updated: now `Editor::open()` returns a bool, as suggested.

I also updated the documentation comment for `Editor::open()` regarding how the parent pointer should be interpreted on different platforms -- requesting feedback from OS X (@robsaunders) and Windows (?) users to make sure the comment is correct.

File: src/editor.rs [hunk 1]
```diff
@@ -11,29 +11,37 @@ pub trait Editor {
...
+    /// `parent` is a window pointer that the new window should attach itself to.
+    /// **It is dependent upon the platform you are targeting.**
+    ///
+    /// A few examples:
+    ///
+    ///  - On Windows, it should be interpreted as a `HWND`
+    ///  - On Mac OS X (64 bit), it should be interpreted as a `NSView*`
+    ///  - On X11 platforms, it should be interpreted as a `u32` (the ID number of the parent window)
+    ///
+    /// Return true if the window opened successfully, false otherwise.
```
Comment (askeksa) [inline src/editor.rs]: Put `true` and `false` in backquotes.

File: src/editor.rs [hunk 2]
```diff
@@ -11,29 +11,37 @@ pub trait Editor {
...
+    fn open(&mut self, parent: *mut c_void) -> bool;
 
     /// Return whether the window is currently open.
     fn is_open(&mut self) -> bool;
 
-
     /// Set the knob mode for this editor (if supported by host).
     ///
     /// Return true if the knob mode was set.
```
Comment (askeksa) [inline src/editor.rs]: Put `true` in backquotes here as well, while you are at it fixing typos and such. :)

File: src/editor.rs [hunk 3]
```diff
@@ -11,29 +11,37 @@ pub trait Editor {
...
     fn set_knob_mode(&mut self, mode: KnobMode) -> bool {
         false
     }
 
-    /// Recieve key up event. Return true if the key was used.
+    /// Receive key up event. Return true if the key was used.
```
Comment (askeksa) [inline src/editor.rs]: Put `true` in backquotes. Also below in `key_down`.

File: src/editor.rs (hunk 1 above)
Comment (crsaracco) [reply src/editor.rs]: Thanks! Changed all of these in the `Editor` trait comments.

**Comment by Walther** [comment]
Tested on
- MacOS Mojave 10.14.1 (18B75)
- https://github.com/crsaracco/rtb-rs-tester at commit `948a3e7`
- Bitwig Studio 2.4.2

The Rutabaga window loads up 🎉 
screen shot 2019-01-19 at 17 36 27
Clicking inside it crashes the plugin 😢 
screen shot 2019-01-19 at 17 36 43
Reloading the plugin works, but requires additional press of the "Show window" toggle. Toggling that indeed shows and hides the plugin window.
screen shot 2019-01-19 at 17 37 05

Then again I was just told in chat that Bitwig ignores the return value, so 🤷‍♂️

**Comment by askeksa** [comment]
Tested on Windows 10 via `rtb-rs-tester` in Renoise, Carla and Bitwig Studio. Window is opening and closing fine in all situations.

**Comment by askeksa** [comment]
What is keeping this from being merged?

**Pull request merged**